var/spack/repos/builtin/packages/flann/package.py

```diff
@@ -0,0 +1,141 @@
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Flann(CMakePackage):
+    """FLANN is a library for performing fast approximate nearest neighbor
+    searches in high dimensional spaces. It contains a collection of
+    algorithms we found to work best for nearest neighbor search and a system
+    for automatically choosing the best algorithm and optimum parameters
+    depending on the dataset.
+
+    FLANN is written in C++ and contains bindings for the following languages:
+    C, MATLAB and Python.
+    """
+
+    homepage = "http://www.cs.ubc.ca/research/flann/"
+    url      = "https://github.com/mariusmuja/flann/archive/1.9.1.tar.gz"
+
+    version('1.9.1', '73adef1c7bf8e8b978987e7860926ea6')
+    version('1.8.5', '02a81640b1e9c11796a0413976dc11f5')
+    version('1.8.4', '774b74580e3cbc5b0d45c6ec345a64ae')
+    version('1.8.1', '1f51500e172f5e11fbda05f033858eb6')
+    version('1.8.0', '473150f592c2997e32d5ce31fd3c19a2')
+
+    # Options available in the CMakeLists.txt
+    # Language bindings
+    variant("python",   default=False,
+            description="Build the Python bindings. "
+                        "Module: pyflann.")
+    variant("matlab",   default=False, description="Build the Matlab bindings.")
+    # default to true for C because it's a C++ library, nothing extra needed
+    variant("c",        default=True,  description="Build the C bindings.")
+
+    # Must build C bindings for Python / Matlab
+    conflicts("+python", when="~c")
+    conflicts("+matlab", when="~c")
+
+    # Additional options
+    variant("cuda",     default=False, description="Build the CUDA library.")
+    variant("examples", default=False, description="Build the examples.")
+    variant("doc",      default=False, description="Build the documentation.")
+    variant("openmp",   default=True,  description="Use OpenMP multi-threading.")
+    # mpi and hdf5 are the bread and butter of this library, use 'em
+    variant("mpi",      default=True,  description="Use MPI.")
+    variant("hdf5",     default=True,  description="Enable HDF5 support.")
+
+    # Dependencies
+    extends("python",      when="+python")
+    depends_on("py-numpy", when="+python", type=("build", "run"))
+    depends_on("matlab",   when="+matlab")
```

Comment:
so. what? are you saying `conflicts("+python", when="+matlab")`?  i don't have `extends("matlab", when="+matlab")` but I can't even test the matlab stuff because I don't have it and refuse to pay for it.  so i guess i'll just leave it?